src/Impl/ConsumerPollingTrait.php

```diff
@@ -0,0 +1,72 @@
+<?php
+declare(strict_types=1);
+
+namespace Interop\Queue\Impl;
+
+use Interop\Queue\Message;
+
+/**
+ * The ConsumerPollingTrait is a common implementation of the polling algorithm for a Consumer
+ */
+trait ConsumerPollingTrait
```

Comment:
I would love to merge this trait and its tests. Could you make a separate PR with only this one?